File: backend/dataset/coinChange/coinChange_20.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        
        min_den  = [(amount+1) for i in range(0,amount+1)]
        min_den[0] = 0
        for i in range(1,len(min_den)):
            min_val = min_den[i]
            
            for j in coins:
                if i-j >=0:
                    min_val = min(min_val, min_den[i-j]+1)
                                        
            min_den[i] = min_val

        if min_den[-1]>amount:
            return -1
        else:
            
            return min_den[-1]

File: backend/CDataset/coinChange/coinChange_20.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
		if (amount == 0) return 0;
		vector<double> result(amount + 1, amount + 1);
		result[0] = 0;
		for (int i = 1; i <= amount; i++)
		{
			for (int j = 0; j < coins.size(); j++)
			{
				if (i >= coins[j] && result[i] - 1 > result[i - coins[j]])
				{
					result[i] = result[i - coins[j]] + 1;
				}
			}
		}
		return (int)result[amount] > amount ? -1 : (int)result[amount];
	}
};
